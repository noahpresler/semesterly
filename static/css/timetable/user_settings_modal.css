.welcome-modal {
	max-width: 500px;
	height: 575px;
	outline: none;;
}

.welcome-modal #modal-header {
	background-color: #eee;
	padding: 25px 25px;
	position: relative;
	border-bottom: 1px solid #ddd;
	margin-bottom: 30px;
}

.welcome-modal .pro-pic {
	border: 2px solid #ddd;
}

.welcome-modal #modal-content {
	padding: 12px;
	text-align: center;
	height: 400px;
}

.welcome-modal #modal-content p {
	color: #aaa;
	margin-bottom: 5px;
}

.welcome-modal #modal-content form {
	text-align: left;
}

.welcome-modal .signup-button {
    bottom: -30px;
    position: fixed;
    margin: 0 auto;
    left: 200px;
}

.welcome-modal h3 {
	text-align: left;
	font-size: 15px;
	margin-bottom: 5px;
	margin-top: 5px;
}

.welcome-modal .Select {
	margin-bottom: 20px;
	margin-top: 10px;
}

.welcome-modal .disclaimer {
	font-size: 11px;
	margin-top: 0px;
	text-align: left;
}

.welcome-modal .switch-slide .switch-label:after {
    background: rgb(90, 200, 251);
}

.preference-wrapper {
	width: 75%;
	float: left;
}

.welcome-modal .switch-slide {
	float: left;
}