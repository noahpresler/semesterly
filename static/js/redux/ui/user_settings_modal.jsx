import React from 'react';
import ReactDOM from 'react-dom';
import fetch from 'isomorphic-fetch';
import { getSchool } from '../init.jsx';
import Modal from 'boron/DropModal';
import Select from 'react-select';
import majors from '../majors.jsx';

export class UserSettingsModal extends React.Component {
    constructor(props) {
        super(props);
        this.changeForm = this.changeForm.bind(this);
        this.changeMajor = this.changeMajor.bind(this);
        this.changeClassYear = this.changeClassYear.bind(this);
        this.shouldShow = this.shouldShow.bind(this);
        this.isIncomplete = this.isIncomplete.bind(this);
    }
    changeForm() {
        let newUserSettings = {
            social_courses: this.refs.share_courses.checked,
            social_offerings: this.refs.share_sections.checked
        }
        let userSettings = Object.assign({}, this.props.userInfo, newUserSettings);
        this.props.changeUserInfo(userSettings);
        this.props.saveSettings();
    }
    componentDidMount() {
        if (this.shouldShow(this.props))
            this.refs.modal.show();
        if(this.isIncomplete(this.props.userInfo.social_courses)) {
            let newUserSettings = { social_courses: true, social_offerings: false };
            let userSettings = Object.assign({}, this.props.userInfo, newUserSettings);
            this.props.changeUserInfo(userSettings);
        }
    }
    componentWillReceiveProps(props) {
        if (this.shouldShow(props))
            this.refs.modal.show();
    }
    changeMajor(val) {
        let userSettings = Object.assign({}, this.props.userInfo, {major: val.value});
        this.props.changeUserInfo(userSettings);
        this.props.saveSettings();
    }
    changeClassYear(val) {
        let userSettings = Object.assign({}, this.props.userInfo, {class_year: val.value});
        this.props.changeUserInfo(userSettings);
        this.props.saveSettings();
    }
    shouldShow(props) {
        return props.userInfo.isLoggedIn && (this.isIncomplete(props.userInfo.social_offerings) || this.isIncomplete(props.userInfo.social_courses) || this.isIncomplete(props.userInfo.major) || this.isIncomplete(props.userInfo.class_year));
    }
    isIncomplete(prop) {
        return prop === undefined || prop === null || prop === "";
    }
    render() {
        return (
            <Modal ref="modal" className="welcome-modal" closeOnClick={false} keyboard={false}>
                <div id="modal-header">
                    <div className="pro-pic" style={{backgroundImage: 'url(' + this.props.userInfo.img_url + ')'}}></div>
                    <h1>Welcome!</h1>
                </div>
                <div id="modal-content">
                    <p>Welcome to Semester.ly, we just have a few quick questions!</p>
                    <h3>What's your major?</h3>
                    <Select
                        name="form-field-name"
                        value={this.props.userInfo.major}
                        ref="major"
                        options={majors}
                        onChange={this.changeMajor}
                    />
                    <h3>What's your graduating class year?</h3>
                    <Select
                        name="form-field-name"
                        value={this.props.userInfo.class_year}
                        ref="class_year"
                        options={[
                            {value: 2013, label: 2013},
                            {value: 2014, label: 2014},
                            {value: 2015, label: 2015},
                            {value: 2016, label: 2016},
                            {value: 2017, label: 2017},
                            {value: 2018, label: 2018},
                            {value: 2019, label: 2019},
                            {value: 2021, label: 2021},
                            {value: 2022, label: 2022},
                            {value: 2023, label: 2023}
                        ]}
                        onChange={this.changeClassYear}
                    />
                    <div className="preference">
                        <div className="preference-wrapper">
                            <h3>Would you like to find classes with friends?</h3>
                            <p className="disclaimer">Find classes with your friends, and allow your friends to find classes with you.</p>
                        </div>
                        <label className="switch switch-slide" onClick={this.changeForm}>
                            <input ref="share_courses" className="switch-input" type="checkbox" checked={this.props.userInfo.social_courses === true}/>
                            <span className="switch-label" data-on="Yes" data-off="No"></span>
                            <span className="switch-handle"></span>
                        </label>
                    </div>
                    <div className="preference">
                        <div className="preference-wrapper">
                            <h3>Would you like to find sections with friends?</h3>
                            <p className="disclaimer">Find specific sections with your friends, and allow your friends to find sections with you.</p>
                        </div>
                        <label className="switch switch-slide" onClick={this.changeForm}>
                            <input ref="share_sections" className="switch-input" type="checkbox" checked={this.props.userInfo.social_offerings === true}/>
                            <span className="switch-label" data-on="Yes" data-off="No"></span>
                            <span className="switch-handle"></span>
                        </label>
                    </div>
                    <button className="signup-button" onClick={() => {
                        this.props.saveSettings();
                        if (!this.shouldShow(this.props))
                                this.refs.modal.hide();
                    }}>Save</button>
                </div>
            </Modal>
        );
    }
}
