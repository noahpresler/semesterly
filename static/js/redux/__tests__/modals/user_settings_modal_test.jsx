/**
Copyright (C) 2017 Semester.ly Technologies, LLC

Semester.ly is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Semester.ly is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
**/

import React from 'react';
import thunk from 'redux-thunk';
import Provider from 'react-redux/src/components/Provider';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import UserSettingsModalContainer from '../../ui/containers/modals/user_settings_modal_container';
import { unfilledFixture, filledFixture, googleFixture } from '../../__fixtures__/user_settings_modal.fixture';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('User Setting Modal Renders As Expected', () => {
  it('VISIBLE if settings unfilled', () => {
    const store = mockStore({
      userInfo: unfilledFixture.userInfo,
      notificationToken: {
        hasToken: false,
      },
      ui: {
        highlightNotifs: false,
      },
    });
    const tree = renderer.create(
      <Provider store={store}><UserSettingsModalContainer /></Provider>,
    ).toJSON();
    delete tree.children[0].children[0].props.style.animationName;
    delete tree.children[1].props.style.animationName;
    expect(tree).toMatchSnapshot();
  });

  it('HIDDEN if settings filled', () => {
    const store = mockStore({
      userInfo: filledFixture.userInfo,
      notificationToken: {
        hasToken: false,
      },
      ui: {
        highlightNotifs: false,
      },
    });
    const tree = renderer.create(
      <Provider store={store}><UserSettingsModalContainer /></Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('VISIBLE if settings filled but showOverrided', () => {
    const userInfo = filledFixture.userInfo;
    userInfo.overrideShow = true;
    const store = mockStore({
      userInfo,
      notificationToken: {
        hasToken: false,
      },
      ui: {
        highlightNotifs: false,
      },
    });
    const tree = renderer.create(
      <Provider store={store}><UserSettingsModalContainer /></Provider>,
    ).toJSON();
    delete tree.children[0].children[0].props.style.animationName;
    delete tree.children[1].props.style.animationName;
    expect(tree).toMatchSnapshot();
  });

  it('VISIBLE but reduced if signing up with Google only', () => {
    const store = mockStore({
      userInfo: googleFixture.userInfo,
      notificationToken: {
        hasToken: false,
      },
      ui: {
        highlightNotifs: false,
      },
    });
    const tree = renderer.create(
      <Provider store={store}><UserSettingsModalContainer /></Provider>,
    ).toJSON();
    delete tree.children[0].children[0].props.style.animationName;
    delete tree.children[1].props.style.animationName;
    expect(tree).toMatchSnapshot();
  });
});
