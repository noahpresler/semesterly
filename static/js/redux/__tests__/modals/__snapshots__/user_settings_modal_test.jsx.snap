// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Setting Modal Renders As Expected HIDDEN if settings filled 1`] = `null`;

exports[`User Setting Modal Renders As Expected VISIBLE but reduced if signing up with Google only 1`] = `
<span>
  <div
    className="welcome-modal max-modal"
    style={
      Object {
        "left": "50%",
        "position": "fixed",
        "top": "50%",
        "transform": "translate3d(-50%, -50%, 0)",
        "width": "100%",
        "zIndex": 1050,
      }
    }
  >
    <div
      style={
        Object {
          "animationDuration": "1s",
          "animationFillMode": "forwards",
          "animationTimingFunction": "linear",
          "backgroundColor": "white",
          "margin": 0,
        }
      }
      tabIndex="-1"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <div
            className="pro-pic"
            style={
              Object {
                "backgroundImage": "url(https://lh5.googleusercontent.com/-jT-TI0NDQHQ/AAAAAAAAAAI/AAAAAAAAQQ8/BM79nJFKph8/photo.jpg?sz=50)",
              }
            }
          />
          <h1>
            Welcome!
          </h1>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="preference cf"
          >
            <h3>
              What's your major?
            </h3>
            <div
              className="Select Select--single is-clearable is-searchable"
              style={undefined}
            >
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-6--value"
                >
                  <div
                    className="Select-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    className="Select-input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-activedescendant="react-select-6--value"
                      aria-describedby={undefined}
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-label={undefined}
                      aria-labelledby={undefined}
                      aria-owns=""
                      className={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "5px",
                        }
                      }
                      tabIndex={undefined}
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </span>
                <span
                  aria-label="Clear value"
                  className="Select-clear-zone"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  title="Clear value"
                >
                  <span
                    className="Select-clear"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&times;",
                      }
                    }
                  />
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <span
                    className="Select-arrow"
                    onMouseDown={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            className="preference cf"
          >
            <h3>
              What's your graduating class year?
            </h3>
            <div
              className="Select Select--single is-clearable is-searchable"
              style={undefined}
            >
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-7--value"
                >
                  <div
                    className="Select-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    className="Select-input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-activedescendant="react-select-7--value"
                      aria-describedby={undefined}
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-label={undefined}
                      aria-labelledby={undefined}
                      aria-owns=""
                      className={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "5px",
                        }
                      }
                      tabIndex={undefined}
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <span
                    className="Select-arrow"
                    onMouseDown={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            className="preference welcome-modal__notifications second cf"
          >
            <button
              className="btn abnb-btn fb-btn"
              onClick={[Function]}
            >
              <span
                className="img-icon"
              >
                <i
                  className="fa fa-facebook"
                />
              </span>
              <span>
                Continue with Facebook
              </span>
            </button>
            <p
              className="disclaimer ctr"
            >
              Connecting your Facebook allows you to see which of your Facebook friends are in your classes! Only friends in your course will see your name â€“ your information is never shared with any other party.
            </p>
          </div>
          <div
            className="preference cf"
          >
            <label
              className="switch switch-slide"
              htmlFor="tos-agreed-input"
            >
              <input
                checked={false}
                className="switch-input"
                disabled={false}
                id="tos-agreed-input"
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="switch-label"
                data-off="CLICK TO ACCEPT"
                data-on="ACCEPTED"
              />
              <span
                className="switch-handle"
              />
            </label>
            <div
              className="preference-wrapper"
            >
              <h3>
                Accept the terms and conditions
              </h3>
              <p
                className="disclaimer"
              >
                By agreeing, you accept our 
                <a>
                  terms and conditions
                </a>
                 & 
                <a>
                  privacy policy
                </a>
                .
              </p>
            </div>
          </div>
          <div
            className="button-wrapper"
          >
            <button
              className="signup-button"
              onClick={[Function]}
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    onClick={null}
    style={
      Object {
        "animationDuration": "0.3s",
        "animationFillMode": "forwards",
        "animationTimingFunction": "linear",
        "backgroundColor": "#373A47",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 1040,
      }
    }
  />
</span>
`;

exports[`User Setting Modal Renders As Expected VISIBLE if settings filled but showOverrided 1`] = `
<span>
  <div
    className="welcome-modal max-modal"
    style={
      Object {
        "left": "50%",
        "position": "fixed",
        "top": "50%",
        "transform": "translate3d(-50%, -50%, 0)",
        "width": "100%",
        "zIndex": 1050,
      }
    }
  >
    <div
      style={
        Object {
          "animationDuration": "1s",
          "animationFillMode": "forwards",
          "animationTimingFunction": "linear",
          "backgroundColor": "white",
          "margin": 0,
        }
      }
      tabIndex="-1"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <div
            className="pro-pic"
            style={
              Object {
                "backgroundImage": "url(https://scontent.xx.fbcdn.net/v/t1.0-1/p50x50/18193968_10208799143997309_4182518417368859549_n.jpg?oh=1e9b37d0a11caa04dc04fd43272fe377&oe=59D65C0A)",
              }
            }
          />
          <h1>
            Welcome!
          </h1>
          <div
            className="modal-close"
            onClick={[Function]}
          >
            <i
              className="fa fa-times"
            />
          </div>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="preference cf"
          >
            <h3>
              What's your major?
            </h3>
            <div
              className="Select Select--single is-clearable is-searchable has-value"
              style={undefined}
            >
              <input
                disabled={false}
                name="form-field-name"
                type="hidden"
                value="Computer Science"
              />
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-4--value"
                >
                  <div
                    className="Select-value"
                    style={undefined}
                    title={undefined}
                  >
                    <span
                      aria-selected="true"
                      className="Select-value-label"
                      id="react-select-4--value-item"
                      role="option"
                    >
                      Computer Science
                    </span>
                  </div>
                  <div
                    className="Select-input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-activedescendant="react-select-4--value"
                      aria-describedby={undefined}
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-label={undefined}
                      aria-labelledby={undefined}
                      aria-owns=""
                      className={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "5px",
                        }
                      }
                      tabIndex={undefined}
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </span>
                <span
                  aria-label="Clear value"
                  className="Select-clear-zone"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  title="Clear value"
                >
                  <span
                    className="Select-clear"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&times;",
                      }
                    }
                  />
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <span
                    className="Select-arrow"
                    onMouseDown={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            className="preference cf"
          >
            <h3>
              What's your graduating class year?
            </h3>
            <div
              className="Select Select--single is-clearable is-searchable has-value"
              style={undefined}
            >
              <input
                disabled={false}
                name="form-field-name"
                type="hidden"
                value="2017"
              />
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-5--value"
                >
                  <div
                    className="Select-value"
                    style={undefined}
                    title={undefined}
                  >
                    <span
                      aria-selected="true"
                      className="Select-value-label"
                      id="react-select-5--value-item"
                      role="option"
                    >
                      2017
                    </span>
                  </div>
                  <div
                    className="Select-input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-activedescendant="react-select-5--value"
                      aria-describedby={undefined}
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-label={undefined}
                      aria-labelledby={undefined}
                      aria-owns=""
                      className={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "5px",
                        }
                      }
                      tabIndex={undefined}
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </span>
                <span
                  aria-label="Clear value"
                  className="Select-clear-zone"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  title="Clear value"
                >
                  <span
                    className="Select-clear"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&times;",
                      }
                    }
                  />
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <span
                    className="Select-arrow"
                    onMouseDown={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              className="preference cf"
            >
              <label
                className="switch switch-slide"
                htmlFor="social-courses-input"
              >
                <input
                  checked={true}
                  className="switch-input"
                  defaultChecked={true}
                  id="social-courses-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="switch-label"
                  data-off="No"
                  data-on="Yes"
                />
                <span
                  className="switch-handle"
                />
              </label>
              <div
                className="preference-wrapper"
              >
                <h3>
                  Would you like to find classes with friends?
                </h3>
                <p
                  className="disclaimer"
                >
                  See which Facebook friends will be your classmates! Only friends in your course will see your name.
                </p>
              </div>
            </div>
            <div
              className="preference cf"
            >
              <label
                className="switch switch-slide"
                htmlFor="share-sections-input"
              >
                <input
                  checked={true}
                  className="switch-input"
                  id="share-sections-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="switch-label"
                  data-off="No"
                  data-on="Yes"
                />
                <span
                  className="switch-handle"
                />
              </label>
              <div
                className="preference-wrapper"
              >
                <h3>
                  Would you like to find sections with friends?
                </h3>
                <p
                  className="disclaimer"
                >
                  See which Facebook friends will be in your section! Only friends in your section will see your name.
                </p>
              </div>
            </div>
            <div
              className="preference cf"
            >
              <label
                className="switch switch-slide"
                htmlFor="social-all-input"
              >
                <input
                  checked={true}
                  className="switch-input"
                  id="social-all-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="switch-label"
                  data-off="No"
                  data-on="Yes"
                />
                <span
                  className="switch-handle"
                />
              </label>
              <div
                className="preference-wrapper"
              >
                <h3>
                  Find new friends in your classes!
                </h3>
                <p
                  className="disclaimer"
                >
                  Find your peers for this semester. All students in your courses will be able to view your name and public Facebook profile.
                </p>
              </div>
            </div>
          </div>
          <div
            className="preference welcome-modal__notifications cf"
          >
            <h3>
              Use Another Browser To Enable Device Notifications
            </h3>
          </div>
          <h3
            className="delete-link"
            onClick={[Function]}
          >
            Delete my account and all related information 
          </h3>
          <div
            className="button-wrapper"
          >
            <button
              className="signup-button"
              onClick={[Function]}
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    onClick={null}
    style={
      Object {
        "animationDuration": "0.3s",
        "animationFillMode": "forwards",
        "animationTimingFunction": "linear",
        "backgroundColor": "#373A47",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 1040,
      }
    }
  />
</span>
`;

exports[`User Setting Modal Renders As Expected VISIBLE if settings unfilled 1`] = `
<span>
  <div
    className="welcome-modal max-modal"
    style={
      Object {
        "left": "50%",
        "position": "fixed",
        "top": "50%",
        "transform": "translate3d(-50%, -50%, 0)",
        "width": "100%",
        "zIndex": 1050,
      }
    }
  >
    <div
      style={
        Object {
          "animationDuration": "1s",
          "animationFillMode": "forwards",
          "animationTimingFunction": "linear",
          "backgroundColor": "white",
          "margin": 0,
        }
      }
      tabIndex="-1"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <div
            className="pro-pic"
            style={
              Object {
                "backgroundImage": "url(https://scontent.xx.fbcdn.net/v/t1.0-1/p50x50/18193968_10208799143997309_4182518417368859549_n.jpg?oh=1e9b37d0a11caa04dc04fd43272fe377&oe=59D65C0A)",
              }
            }
          />
          <h1>
            Welcome!
          </h1>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="preference cf"
          >
            <h3>
              What's your major?
            </h3>
            <div
              className="Select Select--single is-clearable is-searchable"
              style={undefined}
            >
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-2--value"
                >
                  <div
                    className="Select-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    className="Select-input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-activedescendant="react-select-2--value"
                      aria-describedby={undefined}
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-label={undefined}
                      aria-labelledby={undefined}
                      aria-owns=""
                      className={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "5px",
                        }
                      }
                      tabIndex={undefined}
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <span
                    className="Select-arrow"
                    onMouseDown={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div
            className="preference cf"
          >
            <h3>
              What's your graduating class year?
            </h3>
            <div
              className="Select Select--single is-clearable is-searchable"
              style={undefined}
            >
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-3--value"
                >
                  <div
                    className="Select-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    className="Select-input"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-activedescendant="react-select-3--value"
                      aria-describedby={undefined}
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-label={undefined}
                      aria-labelledby={undefined}
                      aria-owns=""
                      className={undefined}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "5px",
                        }
                      }
                      tabIndex={undefined}
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <span
                    className="Select-arrow"
                    onMouseDown={[Function]}
                  />
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              className="preference cf"
            >
              <label
                className="switch switch-slide"
                htmlFor="social-courses-input"
              >
                <input
                  checked={true}
                  className="switch-input"
                  defaultChecked={true}
                  id="social-courses-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="switch-label"
                  data-off="No"
                  data-on="Yes"
                />
                <span
                  className="switch-handle"
                />
              </label>
              <div
                className="preference-wrapper"
              >
                <h3>
                  Would you like to find classes with friends?
                </h3>
                <p
                  className="disclaimer"
                >
                  See which Facebook friends will be your classmates! Only friends in your course will see your name.
                </p>
              </div>
            </div>
            <div
              className="preference cf"
            >
              <label
                className="switch switch-slide"
                htmlFor="share-sections-input"
              >
                <input
                  checked={false}
                  className="switch-input"
                  id="share-sections-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="switch-label"
                  data-off="No"
                  data-on="Yes"
                />
                <span
                  className="switch-handle"
                />
              </label>
              <div
                className="preference-wrapper"
              >
                <h3>
                  Would you like to find sections with friends?
                </h3>
                <p
                  className="disclaimer"
                >
                  See which Facebook friends will be in your section! Only friends in your section will see your name.
                </p>
              </div>
            </div>
            <div
              className="preference cf"
            >
              <label
                className="switch switch-slide"
                htmlFor="social-all-input"
              >
                <input
                  checked={false}
                  className="switch-input"
                  id="social-all-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="switch-label"
                  data-off="No"
                  data-on="Yes"
                />
                <span
                  className="switch-handle"
                />
              </label>
              <div
                className="preference-wrapper"
              >
                <h3>
                  Find new friends in your classes!
                </h3>
                <p
                  className="disclaimer"
                >
                  Find your peers for this semester. All students in your courses will be able to view your name and public Facebook profile.
                </p>
              </div>
            </div>
          </div>
          <div
            className="preference cf"
          >
            <label
              className="switch switch-slide"
              htmlFor="tos-agreed-input"
            >
              <input
                checked={false}
                className="switch-input"
                disabled={false}
                id="tos-agreed-input"
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="switch-label"
                data-off="CLICK TO ACCEPT"
                data-on="ACCEPTED"
              />
              <span
                className="switch-handle"
              />
            </label>
            <div
              className="preference-wrapper"
            >
              <h3>
                Accept the terms and conditions
              </h3>
              <p
                className="disclaimer"
              >
                By agreeing, you accept our 
                <a>
                  terms and conditions
                </a>
                 & 
                <a>
                  privacy policy
                </a>
                .
              </p>
            </div>
          </div>
          <div
            className="button-wrapper"
          >
            <button
              className="signup-button"
              onClick={[Function]}
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    onClick={null}
    style={
      Object {
        "animationDuration": "0.3s",
        "animationFillMode": "forwards",
        "animationTimingFunction": "linear",
        "backgroundColor": "#373A47",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 1040,
      }
    }
  />
</span>
`;
